docs/devel/e2e-tests.md

```diff
@@ -0,0 +1,141 @@
+
+The purpose of *this* document is to serve as a primer for developers who are looking to execute, or add tests, using a local development environment.
+
+## Building and Running the Tests
+
+**NOTE:** The tests have an array of options.  For simplicity, the examples will focus on leveraging the tests on a local cluster using `sudo ./hack/local-up-cluster.sh`
+
+### Building the Tests
+
+The tests are built into a single binary which can be run against any deployed kubernetes system.  To build the tests, navigate to your source directory and execute:
+
+`$ make all`
+
+The output for the end-2-end tests will be a single binary called `e2e.test` under the default output directory, which is typically `_output/local/bin/linux/amd64/`.  Within the repository there are scripts that are provided under the `./hack` directory that are helpful for automation, but may not apply for a local development purposes.  Instead, we recommend familiarizing yourself with the executable options.  To obtain the full list of options, run the following:
+
+`$ ./e2e.test --help`
+
+### Running the Tests
+
+For the purposes of brevity, we will look at a subset of the options, which are listed below:
+
+```
+-ginkgo.dryRun=false: If set, ginkgo will walk the test hierarchy without actually running anything.  Best paired with -v.
+-ginkgo.failFast=false: If set, ginkgo will stop running a test suite after a failure occurs.
+-ginkgo.failOnPending=false: If set, ginkgo will mark the test suite as failed if any specs are pending.
+-ginkgo.focus="": If set, ginkgo will only run specs that match this regular expression.
+-ginkgo.skip="": If set, ginkgo will only run specs that do not match this regular expression.
+-ginkgo.trace=false: If set, default reporter prints out the full stack trace when a failure occurs
+-ginkgo.v=false: If set, default reporter print out all specs as they begin.
+-host="": The host, or api-server, to connect to
+-kubeconfig="": Path to kubeconfig containing embedded authinfo.
+-prom-push-gateway="": The URL to prometheus gateway, so that metrics can be pushed during e2es and scraped by prometheus. Typically something like 127.0.0.1:9091.
+-provider="": The name of the Kubernetes provider (gce, gke, local, vagrant, etc.)
+-repo-root="../../": Root directory of kubernetes repository, for finding test files.
+```
+
+Prior to running the tests, it is recommended that you first create a simple auth file in your home directory, e.g. `$HOME/.kubernetes_auth` , with the following:
+
+```
+{
+  "User": "root",
+  "Password": ""
+}
+```
+
+Next, you will need a cluster that you can test against.  As mentioned earlier, you will want to execute `sudo ./hack/local-up-cluster.sh`.  To get a sense of what tests exist, you may want to run:
```

Comment:
Might want to mention "If you're doing alot of development with local-up-cluster, you may want to clear your genereated credentials and refresh iptables periodically."

- `rm -rf /var/run/kubernetes` (clear kube generated credentials, sometimes stale ones/bad perms can cause problems.
- `sudo iptables -F` (clear ip tables rules, also if doing alot of dev and running local e2e's, this is necessary every so often).